experimental/zerotrie/src/byte_phf/mod.rs

```diff
@@ -0,0 +1,355 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[cfg(feature = "alloc")]
+mod builder;
+#[cfg(feature = "alloc")]
+mod cached_owned;
+
+#[cfg(feature = "alloc")]
+pub use builder::find;
+#[cfg(feature = "alloc")]
+pub use cached_owned::PerfectByteHashMapCacheOwned;
+
+use ref_cast::RefCast;
```

Comment:
issue: imo we should do this manually and not include the extra dep

(we should also consider making this a utility exported by `zerovec_derive` since zerovec situations need it a lot

Reply:
Done